src/Service/Resolvers/CosmosMutationEngine.cs

```diff
@@ -68,6 +80,70 @@ private async Task<JObject> ExecuteAsync(IDictionary<string, object?> queryArgs,
             return response.Resource;
         }
 
+        /// <inheritdoc/>
+        public void AuthorizeMutationFields(
+            IMiddlewareContext context,
+            IDictionary<string, object?> parameters,
+            string entityName,
+            Config.Operation mutationOperation)
+        {
+            string role = string.Empty;
+            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
+            {
+                role = stringVals.ToString();
+            }
+
+            if (string.IsNullOrEmpty(role))
+            {
+                throw new DataApiBuilderException(
+                    message: "No ClientRoleHeader available to perform authorization.",
+                    statusCode: HttpStatusCode.Unauthorized,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed);
+            }
+
+            List<string> inputArgumentKeys;
+            if (mutationOperation != Config.Operation.Delete)
+            {
+                inputArgumentKeys = BaseSqlQueryStructure.GetSubArgumentNamesFromGQLMutArguments(MutationBuilder.INPUT_ARGUMENT_NAME, parameters);
+            }
+            else
+            {
+                inputArgumentKeys = parameters.Keys.ToList();
+            }
+
+            bool isAuthorized; // False by default.
+
+            switch (mutationOperation)
+            {
+                case Config.Operation.UpdateGraphQL:
+                    isAuthorized = _authorizationResolver.AreColumnsAllowedForOperation(entityName, roleName: role, operation: Config.Operation.Update, inputArgumentKeys);
+                    break;
+                case Config.Operation.Create:
+                    isAuthorized = _authorizationResolver.AreColumnsAllowedForOperation(entityName, roleName: role, operation: mutationOperation, inputArgumentKeys);
+                    break;
+                case Config.Operation.Delete:
+                    // Field level authorization is not supported for delete mutations. A requestor must be authorized
+                    // to perform the delete operation on the entity to reach this point.
+                    isAuthorized = true;
+                    break;
+                default:
+                    throw new DataApiBuilderException(
+                        message: "Invalid operation for GraphQL Mutation, must be Create, UpdateGraphQL, or Delete",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest
+                        );
+            }
+
+            if (!isAuthorized)
+            {
+                throw new DataApiBuilderException(
+                    message: "Unauthorized due to one or more fields in this mutation.",
```

Comment:
This message is exactly same as the one we use for SQL, can you create a **const** string in DataBuilderException Class, and use that.

Reply:
updated